File: materijal/10_Fajlovi/primeri/Fajlovi/src/vp/studentskasluzba/Student.java
package vp.studentskasluzba;

public class Student {

	public Student(String brojIndeksa, String ime, String prezime,
			int godinaStudija) {
		this.brojIndeksa = brojIndeksa;
		this.ime = ime;
		this.prezime = prezime;
		this.godinaStudija = godinaStudija;
	}
	
	String brojIndeksa;	
	String ime;
	String prezime;
	int godinaStudija;
	@Override
	public String toString() {
		return "Student [brojIndeksa=" + brojIndeksa + ", ime=" + ime
				+ ", prezime=" + prezime + ", godinaStudija=" + godinaStudija
				+ "]";
	}	
	
	
}

File: materijal/48_JavaScript_ES6/zadaci/resenje/student.js
import Osoba from './osoba.js';

class Student extends Osoba {
    static counter = 100;

    constructor(ime, prezime, polozeniIspiti) {
        super(ime, prezime);
        this.polozeniIspiti = polozeniIspiti || [];
        this.brojIndeksa = `e${Student.counter++}`;
    }

    toString() {
        return `Student: $ { this.id } $ { super.toString() }, $ { this.brojIndeksa }`;
    }

    dodajIspit(ispit) {
        this.polozeniIspiti.push(ispit);
    }

    obrisiIspit(sifraPredmeta) {
        const idx = this.indexOfIspit(sifraPredmeta);
        if (idx !== -1) {
            this.polozeniIspiti.splice(idx, 1);
        }
    }

    izmeniIspit(sifraPredmeta, ocena) {
        const idx = this.indexOfIspit(sifraPredmeta);
        if (idx !== -1) {
            this.polozeniIspiti[idx].ocena = ocena;
        }
    }

    indexOfIspit(sifraPredmeta) {
        return this.polozeniIspiti.findIndex(i => i.predmet.sifra === sifraPredmeta);
    }

    prosecnaOcena() {
        if (!this.polozeniIspiti.length) {
            return 0;
        }

        let sum = this.polozeniIspiti.reduce((accum, val) => accum + val.ocena, 0);
        return sum / this.polozeniIspiti.length;
    }

    ukupnoEspbPoena() {
        return this.polozeniIspiti.reduce((accum, val) => accum + val.predmet.espbPoeni, 0);
    }
}

export default Student;